docs/element-definitions.md

```diff
@@ -234,6 +239,49 @@ Note: At the moment only the HTTP protocol is supported.
 }
 ```
 
+### Affordance (Base API Element)
+
+An affordance element provides a way to provide affordance definitions for the
+API. These definitions can describe link relations used in a Transition Element,
+along with defining unsafe and idempotent transitions.
+
+While data structures are used for defining semantics around data models,
+affrodances are used for defining semantics around hypermedia state transitions.
+
+The control types used here are based on the types used in [ALPS][]. The meta
+property `id` SHOULD be used to define the machine-usable name of the
+affordance.
+
+#### Properties
+
+- `element`: affordance
+- `attributes`
+    - `controlType` (enum)
+        - safe - A hypermedia control that triggers a safe, idempotent state
+          transition (e.g. HTTP.GET or HTTP.HEAD).
+        - unsafe - A hypermedia control that triggers an unsafe, non-idempotent
+          state transition (e.g. HTTP.POST).
+        - idempotent - A hypermedia control that triggers an unsafe, idempotent
+          state transition (e.g. HTTP.PUT or HTTP.DELETE).
+- `content` (array)
+    - (Copy)
+    
+#### Example
+
+```json
+{
+    "element": "affordance",
+    "meta": {
+        "id": "self",
```

Comment:
Since the `id` is to be unique across the whole API Elements document, I wonder if a better name could be used in this example than `self`. Self probably would not be unique across multiple resources defined across API Elements document.